evap/results/tools.py

```diff
@@ -25,30 +25,30 @@
 # see calculate_results
 ResultSection = namedtuple('ResultSection', ('questionnaire', 'contributor', 'label', 'results', 'warning'))
 CommentSection = namedtuple('CommentSection', ('questionnaire', 'contributor', 'label', 'is_responsible', 'results'))
-RatingResult = namedtuple('RatingResult', ('question', 'total_count', 'average', 'deviation', 'counts', 'warning'))
-YesNoResult = namedtuple('YesNoResult', ('question', 'total_count', 'average', 'deviation', 'counts', 'warning', 'approval_count'))
+RatingResult = namedtuple('RatingResult', ('question', 'total_count', 'average', 'counts', 'warning'))
+YesNoResult = namedtuple('YesNoResult', ('question', 'total_count', 'average', 'counts', 'warning', 'approval_count'))
 TextResult = namedtuple('TextResult', ('question', 'answers'))
 HeadingResult = namedtuple('HeadingResult', ('question'))
 
 
-def avg(iterable):
-    """Simple arithmetic average function. Returns `None` if the length of
-    `iterable` is 0 or no items except None exist."""
-    items = [item for item in iterable if item is not None]
-    if len(items) == 0:
-        return None
-    return float(sum(items)) / len(items)
-
+def avg(values, weights=None):
+    """Arithmetic average function. Returns `None` if the length of
+    `values` is 0 or no items except None exist."""
+    if not weights:
+        values = [value for value in values if value is not None]
+        if len(values) == 0:
+            return None
+        return float(sum(values)) / len(values)
 
-def mix(a, b, alpha):
-    if a is None and b is None:
+    average = 0
+    weight_sum = sum([weight for value, weight in zip(values, weights) if value and weight])
+    if weight_sum == 0:
         return None
-    if a is None:
-        return b
-    if b is None:
-        return a
-
-    return alpha * a + (1 - alpha) * b
+    scale = 1 / weight_sum
+    for value, weight in zip(values, weights):
+        if value and weight:
+            average += scale * weight * value
+    return average
```

Comment:
i tried to create a solution that sets the weights to one if the parameter is not present instead of a special case, iterates only once over the values, and crashes on weights that are None (because i think we don't need that do we?). here is what i came up with:
```
def avg(values, weights=itertools.repeat(1)):
    if all(x is None for x in values):
        return None
    
    sum_values = 0
    sum_weights = 0
    for value, weight in zip(values, weights):
        if value:
            sum_values += value * weight
            sum_weights += weight
    return sum_values / sum_weights
```

Reply:
looks good on first sight. although i think `itertools.repeat(1)` doesn't make it easy to understand what we're doing here. i will just copy and paste it because i really can't keep up with all the changes anymore and have not really any idea what the results calculation is currently doing in general. i will review the final thing once you're happy, anything else currently feels like wasted time.

Reply:
well, you can pass weights, and the default for that is a bunch of ones, i thought that was easy to understand...?